cpp/src/strings/like.cu

```diff
@@ -62,7 +88,7 @@ struct like_fn {
 
     auto const target_end  = target_itr + d_str.size_bytes();
     auto const pattern_end = d_pattern.end();
-    auto const esc_char    = d_escape.empty() ? 0 : d_escape[0];
+    auto const esc_char    = d_escape.empty() ? 0 : d_escape.data()[0];
```

Comment:
I'm not sure I grokked this change.  Aren't the two equivalent?

Reply:
No. The original returned a multibyte UTF-8 character and the new one returns just the first byte.